UDV-Core/src/Modules/Temporal/viz.js

```diff
@@ -0,0 +1,86 @@
+import * as vis from 'vis-network';
+
+/**
+* Manager for the graph
+* Done as a singleton so only one instance can exist.
+* If a constructor is called :
+*   - First time = instantiation
+*   - Other time = return the already existing NetworkManagerSingleton
+*/
+export class NetworkManagerSingleton {
+    constructor(start_mode="default",
+                id_network="mynetwork",
+                id_button_location="mybuttons",
+                hidden_button="mode"){
+    /**
+     * Constructor for singleton
+     */
+        const instance = this.constructor.instance;
+        if(instance){
+            return instance;
+        }
+        else{
+            this.network = null;
+            this.data = {"nodes": null,
+                         "edges": null,
+                         "groups": null};
+            this.list_option = null;
+            this.has_changed = true;
+            this.current_mode = start_mode;
+            this.id_network = id_network;
+            this.id_button_location = id_button_location;
+            this.hidden_button = hidden_button;
+
+            this.constructor.instance = this;
+        }
+
+    }
+
+    /**
+     * Kill the simulation network
+     */
+    destroy() {
+        if (this.network !== null) {
+          this.network.destroy();
+          this.network = null;
+        }
+    }
+
+    /**
+    * Initiate the vis.Network with the container (html), data (nodes & edges) and options (graphics)
+    */
+    init(){
+        this.destroy();
+
+        const options = this.list_option[0].visNetwork; // the options could have been multiple
```

Comment:
Why allowing to pass a list if you always take the first ?

Reply:
This has been changed.